zipkin-redis/src/main/scala/com/twitter/zipkin/storage/redis/RedisSnappyCodec.scala

```diff
@@ -0,0 +1,37 @@
+package com.twitter.zipkin.storage.redis
+
+import com.twitter.ostrich.stats.Stats
+import com.twitter.scrooge.{ThriftStruct, ThriftStructSerializer}
+import org.iq80.snappy.Snappy
+import org.jboss.netty.buffer.{ChannelBuffer, ChannelBuffers}
+
+/**
+ * Snappy serializer and deserializer for thrift structs
+ */
+class RedisSnappyThriftCodec[T <: ThriftStruct](val serializer: ThriftStructSerializer[T]) {
+
+  val CompressTimeMetricName = "snappycodec.compress"
+  val CompressBytesMetricName = "redissnappythriftcodec.compressed"
+  val UncompressTimeMetricName = "snappycodec.uncompress"
+  val UncompressBytesMetricName = "redissnappythriftcodec.uncompressed"
+
+  def encode(t: T): ChannelBuffer = {
+    Stats.time(CompressTimeMetricName) {
```

Comment:
not sure we should be doing this in a tight loop. when you are finished testing, mind removing the stats?